packages/payment/src/services/payment-provider.ts

```diff
@@ -0,0 +1,235 @@
+import {
+  Context,
+  CreatePaymentProviderDTO,
+  DAL,
+  InternalModuleDeclaration,
+  IPaymentProcessor,
+  IPaymentProviderService,
+  PaymentProcessorAuthorizeResponse,
+  PaymentProcessorError,
+  PaymentProcessorSessionResponse,
+  PaymentSessionStatus,
+} from "@medusajs/types"
+import {
+  InjectManager,
+  InjectTransactionManager,
+  isPaymentProcessorError,
+  MedusaContext,
+} from "@medusajs/utils"
+import { PaymentProvider } from "@models"
+
+import { isDefined, MedusaError } from "medusa-core-utils"
+import {
+  CreatePaymentInput,
+  PaymentProcessorContext,
+} from "@medusajs/types/src"
+
+type InjectedDependencies = {
+  paymentProviderRepository: DAL.RepositoryService
```

Comment:
todo: missing the providers
```ts
[key: `payment_providers_${string}`]: IPaymentProcessor
```

Reply:
comment: The registration key will be `pp_[string]` in #6311

Reply:
frane, to type accordingly please :)

Reply:
did push the change a while ago but seems like a cache issue with github, didn't see the change on PR until hard refresh